Add namespace-scoped example of UPS Broker

klaudiagrz opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://github.com/kubernetes-incubator/service-catalog/blob/master/CONTRIBUTING.md
2. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
3. See our merge process https://github.com/kubernetes-incubator/service-catalog/blob/master/REVIEWING.md
-->

This PR is a 
 - [X] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**:
In the [walkthrough](https://github.com/kubernetes-incubator/service-catalog/blob/e1ff48f258efe40a3cece8981e22c5d2b72b66ef/docs/walkthrough.md), you show how to register a ClusterServiceBroker basing on the example of UPS Broker. UPS Broker can be either cluster-wide or namespace-scoped. It would be a good idea to add a ServiceBroker example and also update the documentation so that it mentions the possibility to register UPS Broker as a ServiceBroker.


Please leave this checklist in the PR comment so that maintainers can ensure a good PR.

Merge Checklist:
 - [X] New feature 
   - [ ] Tests
   - [ ] Documentation
 - [ ] SVCat CLI flag
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

carolynvs commented:
/ok-to-test

jboyd01 commented:
Hi @klaudiagrz this generally looks good to me, thanks for the contribution.  It may be helpful if you added comments to the top of both of these YAML files that call out the difference between the two (one is for *Cluster* level, one is *Name Space* level) and that fact that if you register both at the same time it will be confusing and difficult to work with the resources as they would both expose resources by the same name but different GUIDs.  

We do have some doc around namespaced brokers:  https://svc-cat.io/docs/namespaced-broker-resources/  do you think we are missing anything or we need to more clearly link to this documentation?

jboyd01 commented on contrib/examples/walkthrough/ups-servicebroker.yaml:
```diff
@@ -0,0 +1,21 @@
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceBroker
+metadata:
+  name: ups-broker
```
If you don't specify the namespace attribute here, it will use the default namespace unless you specify one on the command line for kubectl create.   Might be worth a comment here in here, WDYT?

klaudiagrz commented:
> Hi @klaudiagrz this generally looks good to me, thanks for the contribution. It may be helpful if you added comments to the top of both of these YAML files that call out the difference between the two (one is for _Cluster_ level, one is _Name Space_ level) and that fact that if you register both at the same time it will be confusing and difficult to work with the resources as they would both expose resources by the same name but different GUIDs.

@jboyd01 I've added the comments to both yaml files but actually, I've also renamed them to leave no doubt about which one is which. 

I do not think that the documentation you refer to needs any update. It's really clear and easy to follow 😃

jboyd01 commented:
/approve

klaudiagrz commented:
@carolynvs I was thinking of adding info about this example in the [walkthrough](https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/walkthrough.md) in a form of a note that there is also a possibility to add register a namespace-scoped broker. I can also mention it in the main UPS Broker [README](https://github.com/kubernetes-incubator/service-catalog/blob/master/charts/ups-broker/README.md) in the installation part. What do you think?

carolynvs commented:
Since the example you added is in the walkthrough, I think that's the best place to put a note. Maybe in Step 1, we can note that the walkthrough installs a cluster-wide service broker, and then link to your example of a namespaced service broker.

carolynvs commented:
Oops! The links on our walkthrough page are formatted to display on our jekyll website (preview at: https://deploy-preview-2463--svc-cat.netlify.com/docs/walkthrough/). The links are 404'ing because the yaml file you are linking to is not in the /docs directory. If you switch your links to be absolute URLs to github, then it will work.

Here are the docs for previewing the docs locally (or you can push and view the netlify preview on this PR): https://deploy-preview-2463--svc-cat.netlify.com/docs/devguide/#documentation

jboyd01 commented:
I think we are good on this now.  Thanks for the contribution!
/lgtm

carolynvs commented:
/retest pull-service-catalog-xbuild

carolynvs commented:
/test pull-service-catalog-xbuild

carolynvs commented:
Thanks for the docs improvement, @klaudiagrz! 🎉

klaudiagrz commented:
@jboyd01 @carolynvs Thank you guys for the cooperation! :)
